ui/shared/components/panel/variants/VariantClassify.jsx

```diff
@@ -0,0 +1,115 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { Button, Label } from 'semantic-ui-react'
+import { CopyToClipboard } from 'react-copy-to-clipboard'
+import { HttpRequestHelper } from 'shared/utils/httpRequestHelper'
+import { ColoredIcon } from 'shared/components/StyledComponents'
+import AcmgModal from '../acmg/AcmgModal'
+import { VerticalSpacer } from '../../Spacers'
+import PopupWithModal from '../../PopupWithModal'
+import { getVariantMainTranscript } from '../../../utils/constants'
+
+const CLINGEN_ALLELE_REGISTRY_URL = 'http://reg.genome.network/allele'
+const CLINGEN_VCI_URL = 'https://curation.clinicalgenome.org/select-variant'
+
+class ClinGenAlleleId extends React.PureComponent {
+
+  static propTypes = {
+    hgvsc: PropTypes.string,
+  }
+
+  state = {
+    copied: false,
+    alleleId: null,
+  };
+
+  constructor(props) {
+    super()
+
+    const { hgvsc } = props
+    new HttpRequestHelper(CLINGEN_ALLELE_REGISTRY_URL,
+      (responseJson) => {
+        this.setState({ alleleId: responseJson['@id'].split('/').pop() })
+      },
+      (e) => {
+        this.setState({ alleleId: e.message })
+      }).get({ hgvs: hgvsc }, { credentials: 'omit' })
+  }
+
+  onCopy = () => {
+    this.setState({ copied: true })
+  }
+
+  render() {
+    const { alleleId, copied } = this.state
+
+    return (
+      <CopyToClipboard
+        text={alleleId}
+        onCopy={this.onCopy}
+      >
+        <div>
+          <span>
+            {alleleId}
+            &nbsp;
+          </span>
+          <ColoredIcon name="copy" link />
+          {copied && <ColoredIcon name="check circle" color="#00C000" />}
+        </div>
+      </CopyToClipboard>
+    )
+  }
+
+}
+
+const getButtonBackgroundColor = (classification) => {
+  const categoryColors = {
+    Unknown: 'grey',
+    Benign: 'blue',
+    'Likely Benign': 'blue',
+    Pathogenic: 'orange',
+    'Likely Pathogenic': 'orange',
+    Uncertain: 'yellow',
+  }
+  return categoryColors[classification] || 'grey'
+}
+
+const VariantClassify = React.memo(({ variant, familyGuid }) => {
+  const { hgvsc } = getVariantMainTranscript(variant)
+  const { classify } = variant.acmgClassification || {}
+  const buttonBackgroundColor = getButtonBackgroundColor(classify)
+
+  return (
+    <PopupWithModal
+      content={
+        <div>
+          {hgvsc && (
+            <div>
+              <a href={CLINGEN_VCI_URL} target="_blank" rel="noreferrer">
```

Comment:
As stated in the ticket and in a couple previous reviews, the whole link is not shown until the allele ID is loaded. You need to have a `VCILink` component that handles fetching the allele ID and then showing the link and Allele ID, you cannot have the link directly rendered here

Reply:
I also think about hiding the link before the allele ID is ready. But we may still want the link available even if the ID is unavailable. So I made the link always available. What do you think about it?

Reply:
No. As I stated in 3 separate reviews and the ticket, this is not the behavior we want

Reply:
If we don't show the link while loading, the popup size is smaller, and the `In seqr` is covered by the `loading` spinner, as below figure.
<img src="https://user-images.githubusercontent.com/33550059/218125332-44a5c169-b1f1-4243-b87f-fd4bef48c4eb.png" width=200 height=130 />

After the allele ID is loaded, the popup size grows and covers the button, as shown in the figure below.
<img src="https://user-images.githubusercontent.com/33550059/218125133-2d8d51da-c97c-44c1-80e9-69963eba8b82.png" width=200 height=140 />

So I change it to show the popup under the button as shown below. But it still has the issue of the `In seqr` covered by the spinner before CA id is loaded.
<img src="https://user-images.githubusercontent.com/33550059/218124566-dc5fbf05-4eff-4735-9ef6-ec8e988c4583.png" width=200 height=160 />

Reply:
It looks like the loading spinner is not properly setting its display so the popup is not resizing for it correctly. You can fix this in the CSS, either by fixing the spinner display or by just giving the popup a minimum hight